helper/forwarder-helper.cc

```diff
@@ -79,15 +79,15 @@ ForwarderHelper::InstallPriv (Ptr<Node> node) const
   for (uint32_t i = 0; i < node->GetNDevices (); i++)
     {
       Ptr<NetDevice> currentNetDevice = node->GetDevice (i);
-      if (currentNetDevice->GetObject<LoraNetDevice> () != 0)
+      if (bool(currentNetDevice->GetObject<LoraNetDevice>()) != 0)
```

Comment:
for Ptr<> you can actually use `if(ptr)` directly, there is no need for the cast.
For instance, here you can use  `if (currentNetDevice->GetObject<LoraNetDevice> ())`

Same comment applies for similar modifications

Reply:
Ns-3 currently complains with a deprecation warning if you do an implicit cast from `Ptr<T>` to `bool`. 

```
../contrib/lorawan/model/network-server.cc:91:27: warning: ‘ns3::Ptr<T>::operator ns3::Ptr<T>::Tester*() const [with T = ns3::PointToPointNetDevice]’ is deprecated: see operator bool [-Wdeprecated-declarations]
   91 |       if (p2pNetDevice != 0)
      |                           ^
```
I followed the warning suggestion of using the `bool` operator.
To my knowledge this warning exists at least since ns-3.37.

Reply:
if(ptr) should work, if (ptr != 0) is indeed deprecated

Reply:
Understood, sorry I missed the nuance.

I'll proceed to the change.

Reply:
Thanks, no problem